### ISSUE TITLE
enable the saving and subsequent use of wavefront data along the beamline

### ISSUE THREAD

**Issue by bruhwiler** [radiasoft/sirepo#844]
This is a major source of frustration for users. When designing a beamline, one wants to frequently adjust the parameters of a single element, which often is the last element of an evolving beamline.

The way Sirepo works now, the beamline simulation will start from the source and recalculate the propagation through all of the elements, even if it is only the last element that has been modified.

It should be possible to save the wavefront data before or after any beamline element.
It should also be possible to specify for any intensity plot (or for any watchpoint), that it's data should be generated by starting from the closest set of saved wavefront data.

**Comment by mrakitin** [comment]
I did some progress in this direction. In a customized version of SRW I've implemented reuse or the wavefront data after the last but one element in the beamline before the final drift to later perform multiple propagations in a loop. See commits in my repo https://github.com/mrakitin/multidrift. Currently it works only for a single-electron case. Saving can be easily implemented via pickle, which should be fine for a single machine like Sirepo server, but may not work for execution of the same code in different versions of Python since the format of the pickle files is version specific.

We need to think about the general, code-independent format of exchange of wavefront data for multi-electron cases in SRW to proceed with this issue. HDF5 and other formats to be considered.
